src/mds/CInode.cc

```diff
@@ -4520,7 +4520,14 @@ void CInode::validate_disk_state(CInode::validated_data *results,
       dout(20) << "ondisk_read_retval: " << results->backtrace.ondisk_read_retval << dendl;
       if (results->backtrace.ondisk_read_retval != 0) {
         results->backtrace.error_str << "failed to read off disk; see retval";
-	goto next;
+        // we probably have a new unwritten file!
+        // so skip the backtrace scrub for this entry and say that all's well
+        if (in->is_dirty_parent()) {
```

Comment:
Don't we have enough metadata to be more precise about the file's age rather than accepting any dirty inode with a missing backtrace?